### PR TITLE
refactor(CSRangeHistogram): Refactor csrangehistogram builders

### PR DESCRIPTION
**Pull request by temichus**
- Create public interfaces (functions) and hide internal logic
previously module usage looks like a call of 2 static functions from 2 different classes : 
```
histogram = CSRangeHistogramBuilder.build_from_log_line(log_line=line,
                                                                hst_log_start_time=self.log_start_time)
        summary_data = CSRangeHistogramSummary(histogram).get_summary_for_operation(self.stress_operation)[0]
```

now module has only 3 public functions on the module level. all other stuff is internal. usage is just  call of 1 function:
```
summary_data = make_cs_range_histogram_summary_from_log_line(
            workload=CSWorkloadTypes(self.stress_operation), log_line=line, hst_log_start_time=self.log_start_time)
```
- Improve performance
during profiling was found that the calling method HdrHistogram.add takes about 90% of the running time and internal logic was
build  all HdrHistograms from files  -> make Histogram Summaries based on provided tags tags(WRITE-rt, READ-rt, WRITE-st, READ-st) 

the new algorithm has 2 main improvements:
1) do not build HdrHistograms only for requested tags 
2) parallel "building HdrHistograms ->  make Histogram Summary" work per tag(in make_cs_range_histogram_summary) and "per tag"*"per interval"(in make_cs_range_histogram_summary_by_interval) 

performance test results on SCT runner (2 cores) data from mixed latency 650gb test

start_time=1679355093.7567518, end_time=1679357489.6676207 ~ 2000s
new 68.613 seconds
old 185.282 seconds

start_time=1679355093.7567518, end_time=1679367489.6676207 ~ 12000s
new 313.234 seconds
old 877.644 seconds

so 2.7-2.8 times speed increase in the slow case: latency(according to discussion with @aleksbykov) in small throughput tests speed should be the same due to the histogram file containing only 1 tag

task: https://github.com/scylladb/qa-tasks/issues/1058

## PR pre-checks (self review)

- [ ] I followed [KISS principle](https://en.wikipedia.org/wiki/KISS_principle) and [best practices](https://docs.google.com/document/d/1jihgOKb5iGRlD8_HQ92O0JbLk1kASUoZT23i_MXFSKI)
- [ ] I didn't leave commented-out/debugging code
- [ ] I added the relevant `backport` labels
- [ ] New configuration option are added and documented (in `sdcm/sct_config.py`)
- [ ] I have added tests to cover my changes (Infrastructure only - under `unit-test/` folder)
- [ ] All new and existing unit tests passed (CI)
- [ ] I have updated the Readme/doc folder accordingly (if needed)

### CODE REVIEW

File: sdcm/utils/csrangehistogram.py [hunk 1]
```diff
@@ -64,81 +91,153 @@ def __init__(self, *args, **kwargs):
...
+
+        result = CSRangeHistogramBuilder(None, None, start_interval,
+                                         end_interval).build_histogram_summary_by_tag(path, hdr_tag)
+        if result:
+            return {"interval_num": interval_num, "result": result}
+        return None
+
+    def build_histograms_summary_with_interval(self, path: str, interval=TIME_INTERVAL):  # pylint: disable=too-many-locals
         """
             Build set of time range histogrmas (as list) from
```
Comment (soyacz) [inline sdcm/utils/csrangehistogram.py:242]: this docstring lies now. Please update it.

**Review by soyacz — CHANGES_REQUESTED** [review]
Please add public methods docstring and general `CSRangeHistogramBuilder` docs with link to the feature what this c-s range histogram is.

You dropped some type hints (especially what functions return) and it's not easy to see what is the output (e.g. `build_histogram_summary` returns list of something, while `build_histograms_summary_with_interval_by_tag` returns something else - but both names suggest building the same 'histogram_summary'.

**Comment by roydahan** [comment]
@temichus please describe a bit more in the commit message what was wrong in previous implementation and how the new one is better and add stats you mentioned above.

**Comment by roydahan** [comment]
@fruch, @soyacz & @aleksbykov please review.

**Review by aleksbykov — APPROVED** [review]
LGTM

**Review by soyacz — APPROVED** [review]

**Comment by aleksbykov** [comment]
@soyacz , could you also backport it to perf-v14?

**Comment by soyacz** [comment]
> @soyacz , could you also backport it to perf-v14?

done

**Pull request merged**